### PR TITLE
1446 dag run failed 979s

### LINKED ISSUE #1446: Trigger polling dag for re-running failed digital_bookplate_979 dag runs
When we re-run failed digital_bookplate_979 dag runs (using the airflow UI by clearing the tasks), we need to somehow trigger the poll_for_digital_bookplate_979s_email dag with the dag run ID so that we get an email. The poll_for_digital_bookplate_979s_email dag is triggered with a configuration like:
```
{
    "dag_runs": [
        "manual__2024-10-30T20:38:05.622419+00:00",
        "manual__2024-11-07T19:05:01.569291+00:00",
        "manual__2024-11-07T19:05:01.667437+00:00",
    ],
    "email": null
}
```
Here is a sample request for querying for failed digital_bookplate_979 dag runs:
```
curl -X 'GET' \
  'https://sul-libsys-airflow-dev.stanford.edu/api/v1/dags/digital_bookplate_979/dagRuns?limit=100&state=failed' \
  -H 'accept: application/json'
```
It returns data like this:
```
{
  "dag_runs": [
    {
      "conf": {
        "druids_for_instance_id": {
          "3d67ccc4-5173-47b6-a6cc-011f24c93a65": [
            {
              "druid": "nh023fz6328",
              "fund_name": "HAMMONDA",
              "image_filename": "nh023fz6328_00_0001.jp2",
              "title": "Andrew B. Hammond Memorial Book Fund"
            }
          ]
        }
      },
      "dag_id": "digital_bookplate_979",
      "dag_run_id": "manual__2024-10-30T20:38:05.622419+00:00",
      "data_interval_end": "2024-10-30T20:38:05.622419+00:00",
      "data_interval_start": "2024-10-30T20:38:05.622419+00:00",
      "end_date": "2024-11-18T21:44:32.671466+00:00",
      "execution_date": "2024-10-30T20:38:05.622419+00:00",
      "external_trigger": true,
      "last_scheduling_decision": "2024-11-18T21:44:32.669285+00:00",
      "logical_date": "2024-10-30T20:38:05.622419+00:00",
      "note": null,
      "run_type": "manual",
      "start_date": "2024-11-18T21:43:19.603407+00:00",
      "state": "failed"
    },
...
],
  "total_entries": 14
}
```
In the normal workflow, the polling dag is triggered from the [digital_bookplate_instances dag](https://github.com/sul-dlss/libsys-airflow/blob/787c91915f406206f81c47bdc2011cd53accf88c/libsys_airflow/dags/digital_bookplates/digital_bookplate_instances.py#L110) and from the [upload form](https://github.com/sul-dlss/libsys-airflow/blob/787c91915f406206f81c47bdc2011cd53accf88c/libsys_airflow/plugins/digital_bookplates/apps/digital_bookplates_batch_upload_view.py#L93).

### PR DESCRIPTION
**Pull request by jgreben**
Fixes #1446 

It is not possible to simply update the state of a dag to "QUEUED" (or anything else) and have it run again (see https://github.com/apache/airflow/issues/35729#issuecomment-1819781321). ~~This PR makes it so that a new DAG is launched with the previous DAG's conf. Also, the DAG that finds the previously failed DAGS will only look for ones that are 30 days old, so that it will not keep trying to re-rerun the subsequently failed ones. I made that number 30 days because I vaguely remember that was the period we thought we would run this new DAG. If you want, I can make that number into a Variable or change it for this PR.~~

Using the `airflow.models.DagRun` `clear_dags` method now.

### CODE REVIEW

**Comment by jgreben** [comment]
The draft PR was not actually ready for a review, but it is now. I didn't see a way using the DagRunState model to remove or delete a state.

**Comment by jgreben** [comment]
`https://sul-libsys-airflow-dev.stanford.edu/api/v1/dags/{dag_id}/dagRuns/{dag_run_id}/clear` does work if we want to create a https function instead of using the DagRun class.

**Comment by jgreben** [comment]
> > `https://sul-libsys-airflow-dev.stanford.edu/api/v1/dags/{dag_id}/dagRuns/{dag_run_id}/clear` does work if we want to create a https function instead of using the DagRun class. I also found a clear method in the Dag model https://airflow.apache.org/docs/apache-airflow/stable/_api/airflow/models/dag/index.html#airflow.models.dag.DAG.clear that I can try.
> 
> I thought you could use the [DagRun.set_state()](https://airflow.apache.org/docs/apache-airflow/2.8.3/_api/airflow/models/dagrun/index.html#airflow.models.dagrun.DagRun.set_state) to None but maybe not?

This works:
`dag.clear_dags(
        dags=[dag],
        only_failed=True,
        dry_run=False,
    )`

File: libsys_airflow/plugins/digital_bookplates/dag_979_retries.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+@task
+def failed_979_dags() -> dict:
+    """
+    Find all of the failed digital_bookplate_979 DAG runs
+    """
+    start_date = datetime.now(timezone.utc) - timedelta(30)
+    dag_runs = DagRun.find(
+        state=DagRunState.FAILED,
+        dag_id="digital_bookplate_979",
+        execution_start_date=start_date,
```
Comment (shelleydoljack) [inline libsys_airflow/plugins/digital_bookplates/dag_979_retries.py:25]: Won't we end up not re-running dag runs that failed over 30 days ago?

File: libsys_airflow/plugins/digital_bookplates/dag_979_retries.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    dag_runs = params.get("digital_bookplate_979s", {})
+    logger.info(f"Clearing failed 979 DAG runs: {dag_runs}")
+
+    dagbag = DagBag("/opt/airflow/dags")
+    dag = dagbag.get_dag("digital_bookplate_979")
+    dag.clear_dags(
+        dags=[dag],
+        only_failed=True,
+        dry_run=False,
+    )
```
Comment (shelleydoljack) [inline libsys_airflow/plugins/digital_bookplates/dag_979_retries.py:52]: Does the execution_date get updated when using clear_dags?

**Review by shelleydoljack — APPROVED** [review]
Looks good. I'm not sure about the execution_date for dag runs more than 30 days ago.

File: libsys_airflow/plugins/digital_bookplates/dag_979_retries.py (hunk 2 above)
Comment (jgreben) [reply libsys_airflow/plugins/digital_bookplates/dag_979_retries.py:48]: execution_date removed.

**Review by jermnelson — APPROVED** [review]

**Pull request merged**